htdocs/bom/class/bom.class.php

```diff
@@ -1789,24 +1789,25 @@ class BOMLine extends CommonObjectLine
 	// END MODULEBUILDER PROPERTIES
 
 	/**
-	 * @var float		Calculated cost for the BOM line
+	 * @var float			Calculated cost for the BOM line
 	 */
 	public $total_cost = 0;
 
 	/**
-	 * @var float		Line unit cost based on product cost price or pmp
+	 * @var float			Line unit cost based on product cost price or pmp
 	 */
 	public $unit_cost = 0;
 
 	/**
-	 * @var array     array of Bom in line
+	 * @var BOM				 Reference to Bom in line
 	 */
-	public $childBom = array();
+	public $childBom = null;
 
 	/**
-	 * @var int|null                ID of the unit of measurement (rowid in llx_c_units table)
+	 * * @var int|null		ID of the unit of measurement (rowid in llx_c_units table)
```

Comment:
One '*' too many.

Reply:
Yep I saw that one, I did start on this patch a week or so ago, before your commit came in, it was a bit of a pain to re-integrate and I was doing a bit of copy/paste. It also messed around with the tabs/spaces.

Reply:
The easiest way to handle tabs/spaces is to setup phpcbf which is done automatically when you setup pre-commit.

pip install pre-commit
Then in the local clone: pre-commit install

And pre-commit will run on every commit doing some checks and making some style corrections.

Reply:
Thanks for this, I now have pre-commit  and phpcs/phpcbf